scripts/gen_release_notes/tests/test_helper.py

```diff
@@ -0,0 +1,155 @@
+]
+
+sample_regex_match = [
+    {"regex": r"^helloworld", "context": "helloworld"},
+    {"regex": r"^RC\/0.1", "context": "RC/0.1"},
+    {"regex": r"^0\.1\.2", "context": "0.1.2"},
+    {"regex": r"^some\ faulty\ check", "context": "this is a faulty check"},
+]
+
+sample_regex_search = [
+    {"regex": r"this\ is\ a\ long\ string", "context": "string"},
+    {"regex": r"checkthisrunoutoandonandonandone", "context": "this"},
+    {"regex": r"\[WIP\]\ Releasing tag 0\.10\.10", "context": "0.10.10"},
+    {"regex": r"Nothing\ to\ see\ here", "context": "RC/0.12"},
+]
+
+sample_check_key_dict = [
+    {
+        "dicts": [{"something": "value", "aha": "nice"}, {"another": "value", "confusing": "right"}],
+        "search_key": "another",
+        "search_value": "value",
+    },
+    {
+        "dicts": [{"something": "value", "aha": "nice"}, {"another": "value", "confusing": "right"}],
+        "search_key": "nada",
+        "search_value": "value",
+    },
+    {
+        "dicts": [{"something": "value", "aha": "nice"}, {"another": "value", "confusing": "right"}],
+        "search_key": "value",
+        "search_value": "nice",
+    },
+]
+
+sample_check_valid_env = [
+    {"mod_set": set(), "env_check": "dev", "parse_string": "env/dev/someroot/project"},
+    {"mod_set": {"general"}, "env_check": "prod", "parse_string": "env/prod/anotherroot/project"},
+    {"mod_set": {"dev"}, "env_check": "dev", "parse_string": "module/someenv"},
+    {"mod_set": set(), "env_check": "dev", "parse_string": "module/someenv"},
+]
+
+
+@pytest.mark.parametrize("user_type_changes", sample_type_changes)
+@patch.object(generate, "weight_scale", return_value={"env": 2, "packer": 20})
+def test_get_weight_change(mocked_weight_scale, user_type_changes):
+    """
+    Test to confirm that the weights are expected
+    """
+
+    results = helper.get_weight_change(user_type_changes)
+    if mocked_weight_scale().get(user_type_changes) is not None:
+        assert results == mocked_weight_scale()[user_type_changes]
+    else:
+        assert results == 1
+
+
+@pytest.mark.parametrize("user_string_parser", sample_string_parsers)
+@patch.object(generate, "prd_env_name", return_value="prod")
+@patch.object(generate, "tst_env_name", return_value="tst")
+def test_get_env(mocked_tst_env_name, mocked_prd_env_name, user_string_parser):
+    """
+    Test to confirm that the right env are interpolated
+    """
+
+    results = helper.get_env(user_string_parser["key"])
+    assert results == user_string_parser["expected_value"] and (
+        results == mocked_tst_env_name() or results == mocked_prd_env_name()
+    )
+
+
+@pytest.mark.parametrize("user_regex_match", sample_regex_match)
+def test_is_regex_match(user_regex_match):
+    """
+    Test to confirm the regex match works as intended
+    """
+
+    results = helper.is_regex_match(user_regex_match["regex"], user_regex_match["context"])
+
+    assert results is not None and (results is False or results is True)
```

Comment:
This will always pass no matter the match 😆  I think you want to add the desired outcome for the assert in the parametrize inputs.

Reply:
That was what I initially did actually. I was then reasoning out how the inputs probably shouldn't have logic that determines the outcome of a test. 

However, I think after letting this simmer a bit, I think the path I was thinking if the inputs correspond to some logic in the test that COULD configure a specific var in the test itself. Not the expected end result (if that makes sense).

Either way, I can change it back to what I did previously, it makes this easier to read